File: app/src/main/java/hr/ml/moviesdiscover/util/MovieMapper.java
package hr.ml.moviesdiscover.util;

import java.util.ArrayList;
import java.util.List;

import hr.ml.moviesdiscover.model.Movie;
import hr.ml.moviesdiscover.rest.model.MovieFromRetrofit;

public class MovieMapper {

    public static List<Movie> fromMovieOverviewToMovie(List<MovieFromRetrofit> moviesFromRetrofit) {
        List<Movie> movies = new ArrayList<>();

        for (MovieFromRetrofit movieFromRetrofit :
                moviesFromRetrofit) {

            movies.add(new Movie(movieFromRetrofit.getId(), movieFromRetrofit.getTitle(),
                    movieFromRetrofit.getOriginalLanguage(), movieFromRetrofit.getReleaseDate(),
                    movieFromRetrofit.getPosterPath(), movieFromRetrofit.getVoteAverage(),
                    movieFromRetrofit.getVoteCount()));

        }

        return movies;
    }

}

File: app/src/main/kotlin/hr/ml/moviesdiscover/util/MovieMapper.kt
package hr.ml.moviesdiscover.util

import hr.ml.moviesdiscover.model.Movie
import hr.ml.moviesdiscover.rest.model.MovieFromRetrofit

class MovieMapper {

    companion object {

        fun fromMovieOverviewToMovie(moviesFromRetrofit: List<MovieFromRetrofit>?): List<Movie> {
            val movies = mutableListOf<Movie>()

            val it = moviesFromRetrofit?.iterator()
            while (it?.hasNext() == true) {
                val movieFromRetrofit = it.next()
                movies.add(Movie(movieFromRetrofit.id,
                        movieFromRetrofit.title.orEmpty(),
                        movieFromRetrofit.originalLanguage.orEmpty(),
                        movieFromRetrofit.releaseDate.orEmpty(),
                        movieFromRetrofit.posterPath.orEmpty(),
                        movieFromRetrofit.voteAverage,
                        movieFromRetrofit.voteCount,
                        movieFromRetrofit.overview.orEmpty()))
            }

            return movies
        }

    }

}